src/tab.rs

```diff
@@ -153,13 +156,34 @@ impl EditorTab {
                     text.push_str(line.ending().as_str());
                 }
             });
-            match fs::write(path, text) {
+            match fs::write(path, &text) {
                 Ok(()) => {
                     editor.save_point();
                     log::info!("saved {:?}", path);
                 }
                 Err(err) => {
-                    log::error!("failed to save {:?}: {}", path, err);
+                    if err.kind() == std::io::ErrorKind::PermissionDenied {
+                        log::warn!("Permission denied. Attempting to save with pkexec.");
+
+                        // Start the `pkexec tee` process
+                        if let Ok(mut output) = Command::new("pkexec")
+                            .arg("tee")
```

Comment:
Using `tee` here also writes the file on `cosmic-text`'s stdout. Not sure if there's a better option than `tee` (without install a custom helper to run as root) but it might be good to direct stdout to `Stdio::null()`.

Reply:
Potentially cosmic-edit itself could be used, with special command line arguments. This would allow a custom pkexec rule to be used which could have an appropriate message.